packages/bindgen/CMakeLists.txt

```diff
@@ -193,14 +194,38 @@ if(DEFINED CMAKE_JS_VERSION)
         OUTPUT node_init.cpp
         COMMAND ${NODE_FORCE_COLOR} ${CMAKE_CURRENT_SOURCE_DIR}/realm-bindgen.ts ${SPEC_ARGS} --template node --output ${CMAKE_CURRENT_BINARY_DIR}
         VERBATIM
-        MAIN_DEPENDENCY src/templates/node.ts
+        MAIN_DEPENDENCY src/templates/browser.ts
         DEPENDS
             ${SPEC_FILES}
             SpecJsonSchema
             ${TS_FILES}
     )
 
     target_sources(realm-js PRIVATE node_init.cpp ${CMAKE_JS_SRC} ../../src/node/platform.cpp)
+
+elseif(EMSCRIPTEN)    
+    include(browser.cmake)
+
+    add_custom_command(
+        OUTPUT browser_init.cpp
+        COMMAND ${NODE_FORCE_COLOR} ${CMAKE_CURRENT_SOURCE_DIR}/realm-bindgen.ts ${SPEC_ARGS} --template browser --output ${CMAKE_CURRENT_BINARY_DIR}
+        VERBATIM
+        MAIN_DEPENDENCY src/templates/browser.ts
+        DEPENDS
+            ${SPEC_FILE}
+            SpecJsonSchema
+            ${TS_FILES}
+    )
+
+    target_sources(realm-js PRIVATE browser_init.cpp ${CMAKE_JS_SRC} ../../src/browser/platform.cpp)
+
+    add_custom_command(
+        TARGET realm-js-wasm
+        COMMAND ${CMAKE_COMMAND} -E copy
+                ${CMAKE_CURRENT_BINARY_DIR}/realm-js-wasm.js ${CMAKE_CURRENT_BINARY_DIR}/realm-js-wasm.wasm
+                ${TYPESCRIPT_OUTPUT_DIR}        
+    )
```

Comment:
For the other targets, I've been having them produce their output in the correct output directory directly rather than copying. We should be consistent and either do the same here, or switch the others to copy. I think you need to do something like this change I recently made to node.cmake: https://github.com/realm/realm-js/commit/74f5e2ec6a3ea2b4304ab648c52138879d0b2635#diff-0e67af0e05d8343adec84207c67cc80a294337638fd517c415b7f1b5d72212d0R19